bootstrap_tarball.sh

```diff
@@ -1,168 +1,204 @@
+		rm -f autoopts/option-value-type.c autoopts/option-value-type.h autoopts/optionVersion.3 autoopts/option-xat-attribute.c
+		rm -f autoopts/option-xat-attribute.h autoopts/parse-duration.c autoopts/parse-duration.h autoopts/proto.h
+		rm -f autoopts/save-flags.c autoopts/save-flags.h autoopts/stdnoreturn.in.h autoopts/strequate.3 autoopts/streqvcmp.3
+		rm -f autoopts/streqvmap.3 autoopts/strneqvcmp.3 autoopts/strtransform.3 autoopts/test/Makefile.in
+		rm -f columns/Makefile.in columns/opts.c columns/opts.h
+		rm -f compat/Makefile.in compat/strsignal.h
+		rm -f config/asm-underscore.m4 config/compile config/config.guess config/config.rpath config/config.sub config/depcomp
+		rm -f config/extensions.m4 config/gendocs.sh config/gnulib-cache.m4 config/gnulib-comp.m4 config/guile.m4
+		rm -f config/host-cpu-c-abi.m4 config/install-sh config/lib-ld.m4 config/lib-link.m4 config/lib-prefix.m4 config/libtool.m4
+		rm -f config/ltmain.sh config/lt~obsolete.m4 config/ltoptions.m4 config/ltsugar.m4 config/ltversion.m4 config/missing
+		rm -f config/onceonly.m4 config/pkg.m4 config/snprintfv.m4 config/stdnoreturn.m4 config/test-driver config/texinfo.tex
+		rm -f doc/agdoc.texi doc/autogen.info doc/autogen.info-1 doc/autogen.info-2 doc/autogen.texi doc/gendocs_template
+		rm -f doc/invoke-autogen.texi doc/invoke-bitmaps.texi doc/invoke-columns.texi doc/invoke-getdefs.texi doc/invoke-snprintfv.texi
+		rm -f doc/invoke-xml2ag.texi doc/libopts.texi doc/Makefile.in
+		rm -f getdefs/Makefile.in getdefs/proto.h getdefs/test/defs getdefs/test/Makefile.in
+		rm -f pkg/libopts/stdnoreturn.mk pkg/Makefile.am pkg/Makefile.in
+		rm -f snprintfv/filament.h snprintfv/filament.stamp snprintfv/Makefile.in snprintfv/printf.h snprintfv/printf.stamp
+		rm -f snprintfv/stream.h snprintfv/stream.stamp
+		rm -f xml2ag/fork.c xml2ag/Makefile.in xml2ag/test/Makefile.in xml2ag/xmlopts.c
+
+		# clean up partially generated files
+		sed -n '/\/\* START/,/\/\* END/p' -i agen5/cgi-fsm.c
+
+		# we need to copy some files from git ...
+		cp ../src/agen5/functions.tpl ../src/agen5/Makefile.am.pre agen5
+		cp ../src/autoopts/aoconf.def ../src/autoopts/aoconf.tpl ../src/autoopts/Makefile.am.pre ../src/autoopts/options_h.tpl \
+		   ../src/autoopts/opt-state.def ../src/autoopts/proc-state.def ../src/autoopts/save-flags.def ../src/autoopts/usage-txt.def \
+		   ../src/autoopts/usage-txt.tpl autoopts
+		cp ../src/columns/bootstrap.dir columns
+		cp ../src/getdefs/bootstrap.dir getdefs
+		cp ../src/configure.ac.pre  ../src/NEWS.pre ../src/TODO-top ../src/VERSION.pre .
+		cp ../src/pkg/Makefile.am.pre pkg
+		cp ../src/snprintfv/bootstrap.dir ../src/snprintfv/snprintfv.m4 snprintfv
+
+		# tag the source
+		echo '5.18.987' > ZZJUNK
+		cd ../..
+	fi
+}
+
+bootstrap_columns() {
+	echo "=== Bootstrapping columns ==="
+
+	rm -Rf build/tarball
+	cp -ar build/autogen-5.18.16 build/tarball
+	cd build/tarball
+	patch -p1 -i ../../columns.patch
+	cd columns
+	mkdir -p ../../../build/stage1/bin
```

Comment:
You will need that directory in all three bootstrap paths, won't you? (in case someone only wants to bootstrap getdefs, for example). On the other hand, you cannot bootstrap autogen without getdefs, so maybe it does not need to be added there

Reply:
Yeah, I think it's safe to assume that previous bootstrap stages have to be run by the user (possibly not the download stages if the user is using their own scripting for that but definitely bootstrap stages).

Reply:
I was just thinking that somebody might only need getdefs, which technically does not require columns. But bootstrapping columns is fast, so you don't have to bother.